### PR TITLE
Improved cli syntax highlighting

### LINKED ISSUE #4070: Feature: Support single-line comments in the CLI with syntax highlighting
### Kùzu version

0.5.0

### What operating system are you using?

macOS

### What happened?

Currently, single line comments aren't parsed by the CLI:
```
kuzu> // This is a comment
..> match (n) return n.name;
Error: Parser exception: mismatched input '<EOF>' expecting {ALTER, ATTACH, BEGIN, CALL, CHECKPOINT, COMMENT, COMMIT, COPY, CREATE, DELETE, DETACH, DROP, EXPORT, IMPORT, INSTALL, LOAD, MATCH, MERGE, OPTIONAL, PROJECT, RETURN, ROLLBACK, SET, UNWIND, USE, WITH} (line: 1, offset: 44)
"// This is a comment match (n) return n.name"
```
Multi-line comments work.

```
kuzu> /* This
..> is a comment */
..> match (n:Person) return n.*;
┌────────┐
│ n.name │
│ STRING │
├────────┤
│ Alice  │
│ Bob    │
│ Carol  │
│ Dan    │
└────────┘
(4 tuples)
(1 column)
Time: 0.09ms (compiling), 0.25ms (executing)
```

## No issues in other clients

Kùzu Explorer has no issues and parses the single line comment correctly, presumably because it uses the Node.js client under the hood. The Python client also parses both kinds of comments correctly.

## Desired behaviour

The user should be able to specify single line comments in the CLI that span an entire line, just like they do in the other interfaces to Kùzu.

A simple rule could be "if line starts with `//` parse that entire line as a comment".

### DuckDB CLI supports this

To demonstrate expected behaviour, DuckDB's CLI supports single line comments.

```
❯ duckdb ./ex_db.duckdb
v1.0.0 1f98600c2c
Enter ".help" for usage hints.
D -- This is a comment
D CREATE TABLE person (name STRING PRIMARY KEY);
```

Also, DuckDB takes care of syntax highlighting the comments in the CLI (note how it's grey in colour):
image

### LINKED ISSUE #4167: Shell not highlighting distinct inside count
I tried this query on shell and it does not highlight `distinct`.
```
match (a)-[:fb270Rel]->(b) where a.ID = 6 return count(distinct b);
```

### PR DESCRIPTION
**Pull request by MSebanc**
# Description

Added new syntax highlighting. Highlights keywords, string constants, numbers, and comments. Multiline comments and unclosed brackets/quotes are highlighted in multiline mode. 

image

Fixes #4167, Fixes #4070

# Contributor agreement

- [x] I have read and agree to the [Contributor Agreement](https://github.com/kuzudb/kuzu/blob/master/CLA.md).

### CODE REVIEW

File: tools/shell/linenoise.cpp [hunk 1]
```diff
@@ -2980,8 +3513,8 @@ static char* linenoiseNoTTY(void) {
             }
         }
         int c = fgetc(stdin);
-        if (c == EOF || c == '\n') {
-            if (c == EOF && len == 0) {
+        if (c == -1 || c == '\n') {
```
Comment (andyfengHKU) [inline tools/shell/linenoise.cpp:3516]: Why changing EOF to -1? Is this change related to multiline mode?

**Review by andyfengHKU — APPROVED** [review]
Again I didn't go through each line in details. I just played with the new CLI a bit.

As u mentioned, let's simply merge this PR and gets everyone to try this new feature.

File: tools/shell/linenoise.cpp (hunk 1 above)
Comment (MSebanc) [reply tools/shell/linenoise.cpp]: There were conflicting definitions between the libraries, so I had to change it to -1. I'll make it a variable, so it is a little clearer.
